provider/datagen/src/bin/icu4x-datagen.rs

```diff
@@ -216,6 +231,16 @@ struct Cli {
                 Defaults to 'recommended' for the recommended set of models. Use 'none' for no models"
     )]
     segmenter_models: Vec<String>,
+
+    #[arg(long)]
+    #[arg(help = "Use data from this blob file instead of generating it from sources")]
+    #[cfg(all(feature = "blob_input", feature = "provider"))]
+    input_blob: Option<PathBuf>,
+
+    #[arg(long)]
+    #[arg(help = "Use data from this blob file instead of generating it from sources")]
+    #[cfg(all(feature = "blob_input", not(feature = "provider")))]
+    input_blob: PathBuf,
```

Comment:
Nit: I think I would rather have just a single option `Option<PathBuf>` and `eyre::bail!` if it isn't there when you need it. This is what we do in most other cases where we have CLI options that depend on each other.

Reply:
An `eyre::bail!` gives the nicest messages because we can write them ourselves. It's not just a generic Clap "this argument is required" error. We can explain that the argument is required only because the provider feature is disabled.,